app/src/main/java/org/simple/clinic/drugs/selection/custom/CustomDrugEntryInit.kt

```diff
@@ -7,7 +7,7 @@ import org.simple.clinic.mobius.first
 class CustomDrugEntryInit : Init<CustomDrugEntryModel, CustomDrugEntryEffect> {
   override fun init(model: CustomDrugEntryModel): First<CustomDrugEntryModel, CustomDrugEntryEffect> {
     return when (model.openAs) {
-      is OpenAs.New.FromDrugList -> first(model, FetchDrug(model.openAs.drugUuid), LoadDrugFrequencyChoiceItems)
+      is OpenAs.New.FromDrugList -> first(model.customDrugEntryProgressStateLoading(), FetchDrug(model.openAs.drugUuid), LoadDrugFrequencyChoiceItems)
       is OpenAs.New.FromDrugName -> first(model.drugNameLoaded(model.openAs.drugName), LoadDrugFrequencyChoiceItems)
       is OpenAs.Update -> first(model, FetchPrescription(model.openAs.prescribedDrugUuid), LoadDrugFrequencyChoiceItems)
```

Comment:
I think in general we can probably extract out the model and effect from this when statement, to make it a little bit easier to read?

For example, something like
```kt
val model = when (model.openAs) {
  // handle states
}
```

WDYT?

Reply:
I had thought of this before but this doesn't seem like a better approach in this case.. as we have different effects that we are sending in different cases. So, I'll have to make a when case for effects as well. Which ends up increasing the code than reducing it.

Reply:
Having more lines of code doesn't necessarily mean increased complexity every time, in this scenario we have a couple of repetitions of the same model updates or effects. 

Something like this can be easier to read, or even just making the params in separate lines is also better.

```kt
    val effect = when (model.openAs) {
      is FromDrugList -> FetchDrug(model.openAs.drugUuid)
      is Update -> FetchPrescription(model.openAs.prescribedDrugUuid)
      is FromDrugName -> throw IllegalArgumentException("Should not be called")
    }
    val effects = mutableSetOf(LoadDrugFrequencyChoiceItems, effect)

    val updatedModel = when (model.openAs) {
      is FromDrugList, is Update -> model.customDrugEntryProgressStateLoading()
      else -> model.customDrugEntryProgressStateLoaded()
    }

    return first(updatedModel, effects)
```

Reply:
We can address this in a different PR as well